### DISCUSSION TITLE
How do you clear the message/thread context?

### DISCUSSION THREAD

**Discussion by osseonews** [assistant-ui/assistant-ui#1099] [Q&A] [answered]
We are looking for a simple way to clear the context, so the user can start from scratch with a new thread. I know we can just refresh the page and do that, but that is not a great UX. We were looking thru the hooks to see if there was anything to just reset the thread to clear everything, but couldn't find anything. Any ideas?

**Comment by Yonom** [comment, marked as answer]
depending on your runtime, one of these two should do the trick!
```ts
runtime.reset() // or
runtime.switchToNewThread()
```

you can get the runtime anywhere under AssistantRuntimeProvider via 

```ts
const runtime = useAssistantRuntime()
```

**Reply by osseonews** [reply]
Thanks very much. That worked! BTW, our API sends back the sources of the data used to produce the AI response. Is there any way to pass the sources data from the runtime back to the components  (the sources includes various bits of information about the sources, like a title, link, authors etc)? From what I can tell you can only send back a content array with data (using the custom rest api sample).
